### PR TITLE
github-copilot-cli: init at 0.1.33

### PR DESCRIPTION
**Pull request by malob**
###### Description of changes

Add new package.

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by natsukium** [comment]
"github-copilot-cli" was already packaged in #221090.

**Comment by malob** [comment]
Result of `nixpkgs-review pr 228854` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages failed to build:</summary>
  <ul>
    <li>emojione</li>
    <li>epgstation</li>
    <li>joplin</li>
    <li>lighthouse</li>
  </ul>
</details>
<details>
  <summary>23 packages built:</summary>
  <ul>
    <li>akkoma-frontends.akkoma-fe</li>
    <li>balanceofsatoshis</li>
    <li>bitwarden-cli</li>
    <li>castnow</li>
    <li>commitlint</li>
    <li>create-cycle-app</li>
    <li>emanote</li>
    <li>emanote.data</li>
    <li>github-copilot-cli</li>
    <li>google-clasp</li>
    <li>image_optim</li>
    <li>lv_img_conv</li>
    <li>postcss-cli</li>
    <li>theLoungePlugins.plugins.closepms</li>
    <li>theLoungePlugins.themes.flat-blue</li>
    <li>theLoungePlugins.themes.flat-dark</li>
    <li>thelounge</li>
    <li>uppy-companion</li>
    <li>vimPlugins.coc-stylelint</li>
    <li>vimPlugins.markdown-preview-nvim</li>
    <li>vscode-extensions.rust-lang.rust-analyzer</li>
    <li>wrangler</li>
    <li>zx</li>
  </ul>
</details>

**Comment by malob** [comment]
Result of `nixpkgs-review pr 228854` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package marked as broken and skipped:</summary>
  <ul>
    <li>fast-cli</li>
  </ul>
</details>
<details>
  <summary>2 packages failed to build:</summary>
  <ul>
    <li>epgstation</li>
    <li>joplin</li>
  </ul>
</details>
<details>
  <summary>23 packages built:</summary>
  <ul>
    <li>balanceofsatoshis</li>
    <li>bitwarden-cli</li>
    <li>castnow</li>
    <li>commitlint</li>
    <li>create-cycle-app</li>
    <li>github-copilot-cli</li>
    <li>google-clasp</li>
    <li>haskellPackages.tailwind</li>
    <li>haskellPackages.tailwind.doc</li>
    <li>image_optim</li>
    <li>lighthouse</li>
    <li>lv_img_conv</li>
    <li>postcss-cli</li>
    <li>theLoungePlugins.plugins.closepms</li>
    <li>theLoungePlugins.themes.flat-blue</li>
    <li>theLoungePlugins.themes.flat-dark</li>
    <li>thelounge</li>
    <li>uppy-companion</li>
    <li>vimPlugins.coc-stylelint</li>
    <li>vimPlugins.markdown-preview-nvim</li>
    <li>vscode-extensions.rust-lang.rust-analyzer</li>
    <li>wrangler</li>
    <li>zx</li>
  </ul>
</details>

**Comment by malob** [comment]
Result of `nixpkgs-review pr 228854` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package marked as broken and skipped:</summary>
  <ul>
    <li>fast-cli</li>
  </ul>
</details>
<details>
  <summary>2 packages failed to build:</summary>
  <ul>
    <li>epgstation</li>
    <li>joplin</li>
  </ul>
</details>
<details>
  <summary>28 packages built:</summary>
  <ul>
    <li>balanceofsatoshis</li>
    <li>bitwarden-cli</li>
    <li>castnow</li>
    <li>commitlint</li>
    <li>create-cycle-app</li>
    <li>emanote</li>
    <li>emanote.data</li>
    <li>github-copilot-cli</li>
    <li>google-clasp</li>
    <li>haskellPackages.emanote</li>
    <li>haskellPackages.emanote.data</li>
    <li>haskellPackages.emanote.doc</li>
    <li>haskellPackages.tailwind</li>
    <li>haskellPackages.tailwind.doc</li>
    <li>image_optim</li>
    <li>lighthouse</li>
    <li>lv_img_conv</li>
    <li>postcss-cli</li>
    <li>theLoungePlugins.plugins.closepms</li>
    <li>theLoungePlugins.themes.flat-blue</li>
    <li>theLoungePlugins.themes.flat-dark</li>
    <li>thelounge</li>
    <li>uppy-companion</li>
    <li>vimPlugins.coc-stylelint</li>
    <li>vimPlugins.markdown-preview-nvim</li>
    <li>vscode-extensions.rust-lang.rust-analyzer</li>
    <li>wrangler</li>
    <li>zx</li>
  </ul>
</details>

**Comment by malob** [comment]
@ofborg build nodePackages.@githubnext/github-copilot-cli

**Comment by malob** [comment]
https://github.com/ofborg build nodePackages.@githubnext/github-copilot-cli

**Comment by malob** [comment]
@natsukium, thanks for catching that. Fixed.

File: pkgs/tools/misc/github-copilot-cli/default.nix [hunk 1]
```diff
@@ -0,0 +1,28 @@
+{ lib, buildNpmPackage, fetchzip }:
+
+buildNpmPackage rec {
+  pname = "github-copilot-cli";
+  version = "0.1.33";
+
+  src = fetchzip {
+    url = "https://registry.npmjs.org/@githubnext/${pname}/-/${pname}-${version}.tgz";
+    sha256 = "sha256-uTv6Z/AzvINinMiIfaaqRZDCmsAQ7tOE5SpuecpzGug=";
```
Comment (natsukium) [inline pkgs/tools/misc/github-copilot-cli/default.nix:9]:
```suggestion
    hash = "sha256-uTv6Z/AzvINinMiIfaaqRZDCmsAQ7tOE5SpuecpzGug=";
```
SRI hash is appropriate.

File: pkgs/tools/misc/github-copilot-cli/default.nix [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  npmDepsHash = "sha256-VIg9a63GH246SbmK4Q8CwA2jdaaOwNUXoJkuDVwy5jE=";
+
+  postPatch = ''
+    cp ${./package-lock.json} package-lock.json
+  '';
+
+  dontNpmBuild = true;
+
+  meta = with lib; {
+    description = "A CLI experience for letting GitHub Copilot help you on the command line.";
```
Comment (natsukium) [inline pkgs/tools/misc/github-copilot-cli/default.nix:21]:
```suggestion
    description = "A CLI experience for letting GitHub Copilot help you on the command line";
```

**Review by natsukium — APPROVED** [review]

**Review by wegank — APPROVED** [review]

File: pkgs/tools/misc/github-copilot-cli/default.nix [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+  postPatch = ''
+    cp ${./package-lock.json} package-lock.json
+  '';
+
+  dontNpmBuild = true;
+
+  meta = with lib; {
+    description = "A CLI experience for letting GitHub Copilot help you on the command line";
+    homepage = "https://githubnext.com/projects/copilot-cli/";
+    license = licenses.free;
```
Comment (dotlambda) [inline pkgs/tools/misc/github-copilot-cli/default.nix:23]: Proof?

File: pkgs/tools/misc/github-copilot-cli/default.nix [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+    maintainers = [ maintainers.malo ];
+    platforms = platforms.all;
+  };
+}
+
```
Comment (dotlambda) [inline pkgs/tools/misc/github-copilot-cli/default.nix:28]:
```suggestion
```

File: pkgs/tools/misc/github-copilot-cli/package-lock.json
Comment (dotlambda) [inline pkgs/tools/misc/github-copilot-cli/package-lock.json:1]: Can we please stop vendoring thousands of lines?

Comment (dotlambda) [reply pkgs/tools/misc/github-copilot-cli/package-lock.json:1]: @malob Please ask upstream to distribute a lock file and link to the issue so we can follow the conversation.

Comment (malob) [reply pkgs/tools/misc/github-copilot-cli/package-lock.json:1]: I would have loved to do this. Unfortunately, I wasn't able to find a Git repo I could contribute this change to upstream. As far as I'm aware this is the only location the source is hosted publicly: https://www.npmjs.com/package/@githubnext/github-copilot-cli?activeTab=code

File: pkgs/tools/misc/github-copilot-cli/default.nix (hunk 3 above)
Comment (malob) [reply pkgs/tools/misc/github-copilot-cli/default.nix:23]: This was an oversight on my part. Sorry about that. The package on NPM doesn't contain a license, and not knowing much about how licenses work, I didn't know what to select for the license, so I picked `licenses.free`, and hoped that other folks would point out if that was wrong. Might be a good idea for the `nixpkgs` manual to explicitly say in the [`lib.licenses.unfree`](https://nixos.org/manual/nixpkgs/stable/#lib.licenses.unfree-unfree) that if a package does not have a license explicitly specified, the the package should be marked as `unfree`.

**Pull request merged**